pandas/tests/io/test_excel.py

```diff
@@ -2557,3 +2557,135 @@ def test_excelwriter_fspath(self):
         with tm.ensure_clean('foo.xlsx') as path:
             writer = ExcelWriter(path)
             assert os.fspath(writer) == str(path)
+
+
+@td.skip_if_no('odf')
```

Comment:
Ideally you should add `odf` as an item here:

https://github.com/pandas-dev/pandas/blob/fe1654faa86836a0007bb513504e57c5c9935b8b/pandas/tests/io/test_excel.py#L121

That way all base tests get executed. You might need a skip mark for when things aren't installed. Right now this is done only for xlrd via the class decorator of `SharedItems` but you may need to move that here as well to get this to work

Reply:
I don't understand what the fixture magic is doing here yet.

Reply:
Hmm fair enough. So the fixture is basically allowing all of the SharedItems tests to be run twice, once with an engine explicitly set as `xlrd` and the other with an engine of `None`. 

The inheritance here is kind of tricky because all of the SharedItems operate on actual excel files, which this isn't doing. Not sure the best way to consolidate so tabling this comment for now